src/framework/scene-registry.js

```diff
@@ -376,14 +376,16 @@ class SceneRegistry {
         const app = this._app;
 
         const onBeforeAddHierarchy = (sceneItem) => {
-            // Destroy/Remove all nodes on the app.root
-            const rootChildren = app.root.children;
-            while (rootChildren.length > 0) {
-                const child = rootChildren[0];
-                child.reparent(null);
-                child.destroy?.();
+            // Destroy all nodes on the app.root
+            const { children } = app.root;
+            while (children.length) {
+                children[0].destroy();
             }
-
+            Debug.assert(
```

Comment:
I'm not very familia with the _entityIndex, but could this incorrectly fire if we have entities with GUID that are not added to the root?

Reply:
I can imagine entities that are not under the root .. for example in some pool class, or just stored to be displayed occasionally.

Reply:
Yep, personally I just keep those entities parented and enable/disable when needed, but yes, I can see how that's a possibility.